src/webapp/components/survey/SurveySection.tsx

```diff
@@ -18,19 +18,40 @@ import { Typography } from "@material-ui/core";
 import styled from "styled-components";
 import { QuestionWidget } from "../survey-questions/QuestionWidget";
 import { Question } from "../../../domain/entities/Questionnaire/QuestionnaireQuestion";
+import { useEffect, useState } from "react";
 
 interface SurveySectionProps {
     title: string;
     questions: Question[];
     updateQuestion: (question: Question) => void;
     viewOnly?: boolean;
+    antibioticsBlacklist?: string[];
 }
 export const SurveySection: React.FC<SurveySectionProps> = ({
     title,
     questions,
     updateQuestion,
     viewOnly,
+    antibioticsBlacklist,
 }) => {
+    const [updatedQuestions, setUpdatedQuestions] = useState<Question[]>(questions);
+
+    useEffect(() => {
+        if (antibioticsBlacklist) {
```

Comment:
this is business logic and does not belong in presentation layer.. The question visibility should be set to false in domain layer.

Before the load of any survey, ApplyInitialRulesToSurveyUseCase is called. In the use case you have the module details which will include your antibioticBlackList . Create a function in Questionnaire.ts that contains all biz logic for questionnaire and apply this antibiotic filtering in that function. Call the function from ApplyInitialRulesToSurveyUseCase if antibioticBlackList is applicable to the form. This way all rules logic is in the same use case and easy to debug for future developers.
 We should not have reference of antibioticBlacklist anywhere in the presentation layer.